### ISSUE TITLE
Problem with Xournal++. Logo suddenly disappeared. I have reinstalled.

### ISSUE THREAD

**Issue by LautaroVergara** [xournalpp/xournalpp#5004]
Problem with Xournal++. Logo suddenly disappeared. I have reinstalled and sent this report as suggested when runinning the software.
Thanks.

**Comment by tlacombe** [comment]
Got the same issue. Did re-install fix it?

**Comment by MHoeller** [comment]
Same here, reinstall did not help. It just removed all my settings (Ubuntu 22.04, snap, Version 1.2
Do we, recall that there was once a request, an option to export/import settings ? 
With snaps I didn't made the best experiences when I just copy the config folder..

**Comment by bhennion** [comment]
Which logo are we talking about exactly? Icon in the launcher? Icon in the titlebar of the app? Any other?
And has the icon disappeared (so you can't run the app anymore or so) or has it been replaced by a placeholder?

**Comment by tlacombe** [comment]
As far as I am concerned, it's the icon in the taskbar which as been replaced by the "placeholder gear" of Ubuntu (or gnome, idk). I mention it in [this issue](https://github.com/xournalpp/xournalpp/issues/5018) (completed).

**Comment by bhennion** [comment]
Thanks for the clarification.
The bug is possibly an unintended consequence of #4887

**Comment by bhennion** [comment]
And did you also install it via the snap package?

**Comment by tlacombe** [comment]
> And did you also install it via the snap package?

As far as I am concerned, yes.

**Comment by bhennion** [comment]
Well, I tried to reproduce, but I can't. We'll need more info on your configurations (OS, DE, X11 or Wayland, and so on).
I tried on Archlinux, with the snap package, on KDE and Gnome, X11 and Wayland.

**Comment by makilakixki** [comment]
I can confirm the problem with Xournal++ installed as a snap, in two machines; both running Ubuntu 22.04.3 LTS, one X11 and the other one Wayland.

**Comment by rolandlo** [comment]
I can reproduce the issue when uninstalling all other Xournal++ versions on my system.

**Comment by rolandlo** [comment]
The desktop file in use for the snap package is in `/var/lib/snapd/desktop/applications/xournalpp_xournalpp.desktop`. The StartupWMClass reported (press `ALT+F2` and execute `lg`, then go to Windows tab) is `Com.github.xournalpp.xournalpp` (somehow with a capital `C` in `Com`; I don't know why)

One workaround is the following hack (executed in the terminal):
```term
cd /var/lib/snapd/desktop/applications/
sudo mv xournalpp_xournalpp.desktop Com.github.xournalpp.xournalpp.desktop
sudo sed -i s/com.github/Com.github/g  Com.github.xournalpp.xournalpp.desktop
sudo sed -i s/xournalpp_xournalpp/Com.github.xournalpp.xournalpp/g  Com.github.xournalpp.xournalpp.desktop
```

**Comment by rolandlo** [comment]
Another workaround is to copy the [icon](https://raw.githubusercontent.com/xournalpp/xournalpp/master/ui/pixmaps/com.github.xournalpp.xournalpp.svg) into some folder and use

```term
sudo sed -i s:Icon=:Icon=/path/to/folder/:  /var/lib/snapd/desktop/applications/xournalpp_xournalpp.desktop
```

where you replace `/path/to/folder/` by the absolute path of the folder that contains the icon.

**Comment by rolandlo** [comment]
I opened a [PR](https://github.com/kenvandine/xournalpp/pull/14) on the repository for Xournal++'s snap package, which should fix this issue. I'm closing this issue therefore.

**Comment by michalkvasnicka** [comment]
Is the problem with missing icon or logo fully solved in latest version 1.2.1, or not?
I have the same problem with flatpak install and same problem with PPA: ppa:apandada1/xournalpp-stable
